pkg/services/sqldb/bind.go

```diff
@@ -16,6 +16,15 @@ func (s *serviceManager) ValidateBindingParameters(
 	return nil
 }
 
+//TODO: What behavior do we want for bind on a non-bindable service.
+//Appropriate error?
+func (s *serverOnlyServiceManager) Bind(
+	instance service.Instance,
+	_ service.BindingParameters,
+) (service.BindingContext, service.Credentials, error) {
+	return nil, nil, nil
+}
+
 func (s *serviceManager) Bind(
```

Comment:
Looks like the receiver should be updated on this because the functionality is specific to the `allInOneServiceManager`.

Reply:
I originally had changed it, but reverted it back  because it should work out of the box for the db only mechanism as well. I went with providing new implementations when needed and leaving the base case when it would cover multiple, but I think it would be more explict to change.

Reply:
I think we might be better off keeping types and functions for different `ServiceManager` impls completely separate from one another. Main reason-- as I'd cited in another comment-- is that sharing bits of functionality between `ServiceManagers` in this way forces those `ServiceManagers` into using types that are common to both-- and often types, like contexts, will have fields that vary significantly by service.

When we inevitably end up having to DRY things up due to this approach, we can just have the discreet functions used by each `ServiceManager` delegate to common helper functions.